File: OPENAPI-REST-API/swagger-client/scala-akka-http-server/src/main/scala/io/swagger/server/model/VpsServiceAddons.scala
package io.swagger.server.model


/**
 * Add-on services and IP address information for a VPS, including control panel licenses and extra IP assignments.
 *
 * @param has_cpanel Whether a cPanel license is active on this VPS.
 * @param has_directadmin Whether a DirectAdmin license is active on this VPS.
 * @param has_fantastico Whether a Fantastico license is active on this VPS.
 * @param has_softaculous Whether a Softaculous license is active on this VPS.
 * @param has_hdspace Whether extra disk space has been added to this VPS.
 * @param dedicated_ip Whether a dedicated IP address is assigned to this VPS.
 * @param extra_ips List of additional IPv4 addresses assigned to this VPS.
 * @param extra_ips6 List of additional IPv6 addresses assigned to this VPS.
 * @param unpaid_ips List of IP addresses that have unpaid charges.
 * @param ips All IPv4 addresses assigned to this VPS.
 * @param ips6 All IPv6 addresses assigned to this VPS.
 * @param cpanel_id The add-on service ID for the cPanel license.
 * @param cost Total monthly add-on cost in cents.
 * @param ids List of add-on service IDs active on this VPS.
 * @param rdata Raw add-on data entries.
 */
case class VpsServiceAddons (
  has_cpanel: Option[Boolean],
  has_directadmin: Option[Boolean],
  has_fantastico: Option[Boolean],
  has_softaculous: Option[Boolean],
  has_hdspace: Option[Boolean],
  dedicated_ip: Option[Boolean],
  extra_ips: Option[List[String]],
  extra_ips6: Option[List[String]],
  unpaid_ips: Option[List[String]],
  ips: Option[List[String]],
  ips6: Option[List[String]],
  cpanel_id: Option[Int],
  cost: Option[Int],
  ids: Option[List[String]],
  rdata: Option[List[String]]
)

File: OPENAPI-REST-API/openapi-client/javascript-closure-angular/API/Client/VpsServiceAddons.js
goog.provide('API.Client.VpsServiceAddons');

/**
 * Add-on services and IP address information for a VPS, including control panel licenses and extra IP assignments.
 * @record
 */
API.Client.VpsServiceAddons = function() {}

/**
 * Whether a cPanel license is active on this VPS.
 * @type {!boolean}
 * @export
 */
API.Client.VpsServiceAddons.prototype.hasCpanel;

/**
 * Whether a DirectAdmin license is active on this VPS.
 * @type {!boolean}
 * @export
 */
API.Client.VpsServiceAddons.prototype.hasDirectadmin;

/**
 * Whether a Fantastico license is active on this VPS.
 * @type {!boolean}
 * @export
 */
API.Client.VpsServiceAddons.prototype.hasFantastico;

/**
 * Whether a Softaculous license is active on this VPS.
 * @type {!boolean}
 * @export
 */
API.Client.VpsServiceAddons.prototype.hasSoftaculous;

/**
 * Whether extra disk space has been added to this VPS.
 * @type {!boolean}
 * @export
 */
API.Client.VpsServiceAddons.prototype.hasHdspace;

/**
 * Whether a dedicated IP address is assigned to this VPS.
 * @type {!boolean}
 * @export
 */
API.Client.VpsServiceAddons.prototype.dedicatedIp;

/**
 * List of additional IPv4 addresses assigned to this VPS.
 * @type {!Array<!string>}
 * @export
 */
API.Client.VpsServiceAddons.prototype.extraIps;

/**
 * List of additional IPv6 addresses assigned to this VPS.
 * @type {!Array<!string>}
 * @export
 */
API.Client.VpsServiceAddons.prototype.extraIps6;

/**
 * List of IP addresses that have unpaid charges.
 * @type {!Array<!string>}
 * @export
 */
API.Client.VpsServiceAddons.prototype.unpaidIps;

/**
 * All IPv4 addresses assigned to this VPS.
 * @type {!Array<!string>}
 * @export
 */
API.Client.VpsServiceAddons.prototype.ips;

/**
 * All IPv6 addresses assigned to this VPS.
 * @type {!Array<!string>}
 * @export
 */
API.Client.VpsServiceAddons.prototype.ips6;

/**
 * The add-on service ID for the cPanel license.
 * @type {!number}
 * @export
 */
API.Client.VpsServiceAddons.prototype.cpanelId;

/**
 * Total monthly add-on cost in cents.
 * @type {!number}
 * @export
 */
API.Client.VpsServiceAddons.prototype.cost;

/**
 * List of add-on service IDs active on this VPS.
 * @type {!Array<!string>}
 * @export
 */
API.Client.VpsServiceAddons.prototype.ids;

/**
 * Raw add-on data entries.
 * @type {!Array<!string>}
 * @export
 */
API.Client.VpsServiceAddons.prototype.rdata;

